src/libraries/Microsoft.Extensions.Logging/src/LoggerFactoryScopeProvider.cs

```diff
@@ -185,7 +220,53 @@ IEnumerator IEnumerable.GetEnumerator()
                 return GetEnumerator();
             }
         }
+
+        private class AdditionalActivityItemsLogScope<TValue> : IEnumerable<KeyValuePair<string, TValue>>
+        {
+            private readonly IEnumerable<KeyValuePair<string, TValue?>> _additionalItems;
+
+            public AdditionalActivityItemsLogScope(IEnumerable<KeyValuePair<string, TValue?>> additionalItems)
+            {
+                _additionalItems = additionalItems;
+            }
+
+            public IEnumerator<KeyValuePair<string, TValue>> GetEnumerator()
+            {
+                return _additionalItems.GetEnumerator();
+            }
+
+            IEnumerator IEnumerable.GetEnumerator()
+            {
+                return _additionalItems.GetEnumerator();
+            }
+
+            public override string ToString()
+            {
+                IEnumerator<KeyValuePair<string, TValue?>> enumerator = _additionalItems.GetEnumerator();
+                if (!enumerator.MoveNext())
+                {
+                    return string.Empty;
+                }
+
+                StringBuilder sb = new StringBuilder();
```

Comment:
Unsure about this.
Could/should we cache this sb? I e.g. saw in the ConsoleLogger queue that the SB is marked as thread local and reused.